backend/fetcher.py

```diff
@@ -86,7 +86,9 @@ def fetch_logs(self) -> list[dict[str, str]]:
 
         if self.chroot == "srpm-builds":
             build = self.client.build_proxy.get(self.build_id)
-            baseurl = os.path.dirname(build.source_package["url"])
+            baseurl = None
```

Comment:
Looking at the issue, I think it may be better to rely on the `get` method of the dict.
For example:

```
baseurl = os.path.dirname(build.source_package.get("url", ""))
```
That way we can catch the issue with appropriate exception on line 97/99